compiler/src/dotty/tools/dotc/typer/ImportSuggestions.scala

```diff
@@ -249,6 +252,45 @@ trait ImportSuggestions:
     finally timer.cancel()
   end importSuggestions
 
+  /** The `ref` parts of this list of pairs, discarding subsequent elements that
+   *  have the same String part. Elements are sorted by their String parts.
+   */
+  def (refs: List[(TermRef, String)]).distinctRefs(using Context): List[TermRef] = refs match
+    case (ref, str) :: refs1 =>
+      ref :: refs1.dropWhile(_._2 == str).distinctRefs
+    case Nil =>
+      Nil
+
+  /** The best `n` references in `refs`, according to `compare`
+   *  `compare` is a partial order. If there's a tie, we take elements
+   *  in the order thy appear in the list.
+   */
+  def (refs: List[TermRef]).best(n: Int)(using Context): List[TermRef] =
+    val top = new Array[TermRef](n)
+    var filled = 0
+    val rest = new mutable.ListBuffer[TermRef]
+    for ref <- refs do
+      var i = 0
+      var diff = 0
+      while i < filled && diff == 0 do
+        diff = compare(ref, top(i))(using ctx.retractMode(Mode.ImplicitsEnabled))
```

Comment:
Store `ctx.retractMode(Mode.ImplicitsEnabled)` in a val to avoid recomputing it at every iteration

Reply:
@smarter could you apply this fix directly, as per `fasttrack` label policy?